packages/firestore/test/unit/core/query.test.ts

```diff
@@ -614,4 +616,33 @@ describe('Query', () => {
     query = baseQuery.withEndAt(bound([], true));
     expect(query.matchesAllDocuments()).to.be.false;
   });
+
+  it('can handle conversion to Targets from mirror queries', () => {
+    const limitQuery = Query.atPath(ResourcePath.fromString('collection'))
+      .addOrderBy(orderBy('foo'))
+      .addOrderBy(orderBy('bar', 'desc'))
+      .withStartAt(bound([['foo', 'foo', 'asc']], true))
+      .withLimitToFirst(10);
+    const fromLimit = limitQuery.toTarget();
+
+    const limitToLastQuery = Query.atPath(ResourcePath.fromString('collection'))
+      .addOrderBy(orderBy('foo', 'desc'))
+      .addOrderBy(orderBy('bar', 'asc'))
+      .withEndAt(bound([['foo', 'foo', 'desc']], false))
+      .withLimitToLast(10);
+    const fromLimitToLast = limitToLastQuery.toTarget();
+
+    expect(fromLimit.canonicalId()).to.equal(fromLimitToLast.canonicalId());
+    expect(fromLimit.isEqual(fromLimitToLast)).to.be.true;
+
+    expect(limitQuery.isEqual(fromLimit.toTargetQuery())).to.be.true;
+    expect(limitQuery.canonicalId()).to.equal(
+      fromLimit.toTargetQuery().canonicalId()
+    );
+
+    expect(limitQuery.isEqual(fromLimitToLast.toTargetQuery())).to.be.true;
```

Comment:
I am still concerned about this method. The contract just doesn't feel right and seeing this spelled out in a test makes this obvious. I understand that there may be a need for this conversion, but since this is limited to Multi-Tab and Web-only, what to you think of handling this conversion in a private method in Sync Engine (or a free-standing function):

```
function convertTargetToMultiTabQuery(...)
```

Reply:
Good point that this is multi-tab only. Moved it to SyncEngine.